### DISCUSSION TITLE
[@types\cluster] In the cluster.d.ts, the default export is used, resulting in the default attribute that should not appear after compilation.

### DISCUSSION THREAD

**Discussion by NaOHCC** [DefinitelyTyped/DefinitelyTyped#59229] [Issues with a @types package]
```typescript
import cluster from "cluster";
import { cpus } from "os";

cluster.setupPrimary({
    exec: "worker.js",
});
let numCPUs = cpus().length;
for (let i = 0; i < numCPUs; i++) {
    cluster.fork();
}
```
after compiler
```javascript
Object.defineProperty(exports, "__esModule", { value: true });
const cluster_1 = require("cluster");
const os_1 = require("os");
cluster_1.default.setupPrimary({
    exec: "worker.js",
});
let numCPUs = (0, os_1.cpus)().length;
for (let i = 0; i < numCPUs; i++) {
    cluster_1.default.fork();
}
```
run these codes, Node gives an error
```
cluster_1.default.setupPrimary({
                  ^

TypeError: Cannot read properties of undefined (reading 'setupPrimary')
```
I went to check the TypeScript definition file @types\node\cluster.d.ts, it is different from other files, it uses default.
```typescript
declare module 'cluster' {
// ...
    const cluster: Cluster;
    export default cluster;
}
```

[a comment by an automated account is not shown]

**Reply by ronjouch** [reply]
Confirmed, `cluster` 16.x types are unusable:

- DT types want you to use a default export...
- ... but the module itself does *not* export as default module!

See [TypeScript playground](https://www.typescriptlang.org/play?esModuleInterop=false&declaration=false&experimentalDecorators=false&emitDecoratorMetadata=false&target=8&jsx=0&module=1&ssl=1&ssc=1&pln=2&pc=1#code/PTAEBUAsEsGdTqAhqAxgewA4E8C0mlYAXFdAM1ABFxhUAbAV2IFMAnUI7TZ+JI0SESKZYALhABzaEUgMARgDoMAW2CVmZaADtpzOtnBdmAEzUbtu-Ye6m5ddHODLCRNsE7dYwLemPNajCysCsYKRLAAxAAyAIwADLhRAMwA7ABQIBzooH7K6FrErHzMAugA7hyQJazMKsrMWn7GoMwAHkjKmHQlxug8WgDk-GT6oGXSkBAAyqBM2hJo6Mp5WgBWsAoZYFU1oAAUSI2gcnrlAJQANMhkruxIxsbzlSUARABUyPAvoGTo7D5aXB+MhIBh0fjQTp-cIXLagACSAnuWVAdGKdweoAAAuFcG1uKgiHjWKw-iiJMx+OAZkRsipMNBuqBRLgwCy2ay0pDMNDQB9CGhArcfqTlKABvQmLcBgBuLlQ1j8fnwQTCEVLcWqzCy+U8xWgADeaEwTFAAF91WKBuhYDrubzlaBMKTUDx4GRReLnehXbBbXK0hgCvwtAxlABhAAKAFV4ABeY1MPZnBTdLQSGQBzI42B41oEolsUnsdlcih7SVBBRwSOsSFIVjYM6GtKgRYFdDdVPoCR7AAGtfrjdAABIDd7fRsGcYLYhWAwtDp032zgG25kAGJ-ADWYx3bA2rZ+ZL23QhoATcRlCFAAB5QKGIzHYNfoABqN-Ng1H9dgHN5gtiWLZlOTbNtK1uBRflYbdkzlNszTSI9s1xfFakLEkyVLcChTYBR8j2AY2mkAYrj2Mp91YK4MD8K5YGgCQtCQOhmzjAA+FswPbWBO2Ybtez7CiYLYUcDSE7c8InN0FGnC1HhMFd4PNVc0gtPRYBKb9f1AAB1Sj4FQQ5QFgSAGxKQ5sAgKN2y0dDoHyZCwHhLRKkQQyNIQCFeBcgAJcBwEjYy2AANzYI8tSUGpiimEK2D2PYagARyuGpYFYjitLA1KFDKOtXB85h7j2AAmOI4hUrjsoaYxCKqOh7D3Vg6GMAAdQYKuU1M4FcLQ9gADjK8q1243j+P7PThPYMcpL9GToBnYySEVBSVLNIA) and [StackOverflow: How to use Node.js `cluster` module with DT @types/node@16.x and TypeScript/commonjs?](https://stackoverflow.com/questions/72548137/how-to-use-node-js-cluster-module-with-dt-types-node16-x-and-typescript-comm).

Past maintainers @SimonSchick @sandersn @y-i @jablko, can you halp?

**Comment by kirrg001** [comment]
Also running into this. Getting TS errors for cluster module. I have no idea how to fix this other than adding 

> // @ts-expect-error :-/

Node v18.14.2
TS ^4.9.5
@types/node ^18.14.2

```
const cluster = require('cluster')
cluster.setupPrimary({
    exec: 'worker.js',
    args: ['--use', 'http'],
})
console.log(cluster.settings)
```

```
src/index.js:3:9 - error TS2339: Property 'setupPrimary' does not exist on type 'typeof import("cluster")'.

3 cluster.setupPrimary({
          ~~~~~~~~~~~~

src/index.js:8:21 - error TS2339: Property 'settings' does not exist on type 'typeof import("cluster")'.

8 console.log(cluster.settings)
```
